nilearn/regions/tests/test_hierarchical_kmeans_clustering.py

```diff
@@ -128,3 +129,26 @@ def test_hierarchical_k_means_clustering():
     assert_array_almost_equal(X_compress, X_compress_scaled)
 
     del X_red, X_compress, X_red_scaled, X_compress_scaled
+
+
+@pytest.mark.parametrize("n_clusters", [2, 4, 5])
+def test_hierarchical_k_means_clustering_surface(
+    surf_img, surf_mask, n_clusters
+):
+    """Test hierarchical k-means clustering on surface."""
+    # create a surface masker
+    masker = SurfaceMasker(surf_mask()).fit()
+    # mask the surface image with 50 samples
+    X = masker.transform(surf_img((50,))).T
+    # instantiate HierarchicalKMeans with n_clusters
+    clustering = HierarchicalKMeans(n_clusters=n_clusters)
+    # fit and transform the data
+    X_transformed = clustering.fit_transform(X)
+    # inverse transform the transformed data
+    X_inverse = clustering.inverse_transform(X_transformed)
+
+    # make sure the n_features in transformed data were reduced to n_clusters
+    assert X_transformed.shape[0] == n_clusters
+
+    # make sure the inverse transformed data has the same shape as the original
+    assert X_inverse.shape == X.shape
```

Comment:
> You probably want to have a test running Ward's clustering in the surface ?

Did you mean HierarchicalKmeansClustering? If so, now testing it here.

Reply:
Kmeans or the hierarchical version do not use the spatial structure (mesh). Ward does.

Reply:
While doing so, I noticed that `HierarchicalKMeans.fit()` takes transposed X:
https://github.com/nilearn/nilearn/blob/135e5388386c24ca879328040273127f8acb4a6a/nilearn/regions/hierarchical_kmeans_clustering.py#L233

See in `ReNA.fit()` for comparison:
https://github.com/nilearn/nilearn/blob/135e5388386c24ca879328040273127f8acb4a6a/nilearn/regions/rena_clustering.py#L518

Is this a bug or a feature?